js/views/buyWizardVw.js

```diff
@@ -165,14 +162,12 @@ module.exports = Backbone.View.extend({
     loadTemplate('./js/templates/buyWizard.html', function(loadedTemplate) {
       self.$el.html(loadedTemplate(self.model.toJSON()));
 
-      //append the view to the passed in parent
-      self.parentEl.append(self.$el);
-
       //add subviews
       self.buyDetailsView = new buyDetailsVw({model: self.model});
```

Comment:
Any view that you are creating in render should be removed first if it exists, e.g:
`self.buyDetailsView && self.buyDetailsView.remove()`

The base view will only remove children when the view is removed, but if you're re-rendering multiple times in the views lifecycle, you'll be queueing up child views.